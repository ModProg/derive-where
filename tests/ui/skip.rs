use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Debug)]
struct DuplicateSkipSame<T>(#[derive_where(skip, skip)] PhantomData<T>);

#[derive_where(Debug)]
struct DuplicateSkipSeparate<T>(
	#[derive_where(skip)]
	#[derive_where(skip)]
	PhantomData<T>,
);

#[derive_where(Debug)]
struct EmptySkip<T>(#[derive_where(skip())] PhantomData<T>);

#[derive_where(Debug)]
struct OverridingSkip<T>(
	#[derive_where(skip)]
	#[derive_where(skip(Debug))]
	PhantomData<T>,
);

#[derive_where(Debug)]
struct UnderridingSkip<T>(
	#[derive_where(skip(Debug))]
	#[derive_where(skip)]
	PhantomData<T>,
);

#[derive_where(Clone; T)]
struct NoSupportedTrait<T>(#[derive_where(skip)] PhantomData<T>);

#[derive_where(Debug)]
struct DuplicateTraitSame<T>(#[derive_where(skip(Debug, Debug))] PhantomData<T>);

#[derive_where(Debug)]
struct DuplicateTraitSeparate<T>(
	#[derive_where(skip(Debug))]
	#[derive_where(skip(Debug))]
	PhantomData<T>,
);

#[derive_where(Clone; T)]
struct MissingDeriveTrait<T>(#[derive_where(skip(Debug))] PhantomData<T>);

#[derive_where(Clone, Copy)]
struct SkipCloneWhileCopy<T>(#[derive_where(skip(Clone))] PhantomData<T>);

fn main() {}
