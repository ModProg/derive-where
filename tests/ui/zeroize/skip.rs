use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Skip; T)]
struct UnsupportedTrait<T>(#[derive_where(skip(Skip))] PhantomData<T>);

fn main() {}
