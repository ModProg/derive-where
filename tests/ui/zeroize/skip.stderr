error: unsupported trait, expected one of Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd, Zeroize, ZeroizeOnDrop
 --> tests/ui/zeroize/skip.rs:5:16
  |
5 | #[derive_where(Skip; T)]
  |                ^^^^
